yoyodyne/datasets.py

```diff
@@ -464,6 +470,9 @@ def _iter_samples(
                 features = self._get_cell(
                     row, self.features_col, self.features_sep
                 )
+                # Use unique encoding for features.
+                # This disambiguates from overlap with source.
+                features = [f"<{f}>" for f in features]
```

Comment:
We should use some other style for this, as it could easily clash with the other special characters (for instance if a feature name is `S` or `E`). I am going to suggest square brackets. Then the unknown feature is literally just `<UNK>` in feature form, namely `[<UNK>]`